src/codewhisperer/util/runtimeLanguageContext.ts

```diff
@@ -158,10 +158,25 @@ export class RuntimeLanguageContext {
     }
 
     /**
-     * @param languageId : vscode language id or codewhisperer language name
+     * @param languageId : vscode language id or codewhisperer language name, fileExtension: extension of the selected file
      * @returns An object with a field language: CodewhispererLanguage, if no corresponding CodewhispererLanguage ID, plaintext is returned
      */
-    public getLanguageContext(languageId?: string): { language: CodewhispererLanguage } {
+    public getLanguageContext(languageId?: string, fileExtension?: string): { language: CodewhispererLanguage } {
+        const extensionToLanguageMap: Record<string, CodewhispererLanguage> = {
+            tf: 'tf',
+            hcl: 'tf',
+            json: 'json',
+            yaml: 'yaml',
+            yml: 'yaml',
+            // Add more mappings if needed
+        }
+
+        if (languageId === 'plaintext' && fileExtension !== undefined) {
+            const languages = extensionToLanguageMap[fileExtension]
+            if (languages) {
+                return { language: languages }
+            }
```

Comment:
```
public getLanguageContext(doc: vscode.TextDocument): CodewhispererLanguage {
  return this.normalizeLanguage(doc.languageId) ?? extensionToLanguageMap[fileExt] ?? 'plaintext'
}
```

Reply:
We can not use `(doc: vscode.TextDocument)` as parameter because this `getLanguageContext` is used in different parts of code, example in this we pass `request: codewhispererClient.ListRecommendationsRequest | codewhispererClient.GenerateRecommendationsRequest` so we can not generalize into document as parameter
```      
                runtimeLanguageContext.getLanguageContext(
                    request.fileContext.programmingLanguage.languageName,
                    request.fileContext.filename.substring(request.fileContext.filename.lastIndexOf('.') + 1)
                ).language
```

Reply:
I see.